src/test/test-strv.c

```diff
@@ -407,6 +407,29 @@ static void test_strv_split_extract(void) {
         assert_se(streq_ptr(l[5], NULL));
 }
 
+static void test_strv_split_extract_pairs_colon(void) {
+        _cleanup_strv_free_ char **l = NULL;
+        const char *str = "one:two three four:five six seven:eight";
```

Comment:
Maybe add `"seven:eight:nine"` and verify that it results in `"seven"` and `"eight:nine"` in the array?

Reply:
Good suggestion - but ```seven:eight:nine``` results in ```seven``` and ```eight```, with ```nine``` being discarded. I think this is the behaviour we want for these kind of options? Or do we want to allow using the separator in the last member?

The current behaviour has the nice property of being extendable in the future - say we want to add a third element, we can do it in a backward-compatible way. If ```eight:nine``` would be allowed as a single last value, then it wouldn't be possible.

Reply:
This is now changed as suggested